test/transform.js

```diff
@@ -55,3 +60,77 @@ test('transform should work with a Function', t => {
     t.doesNotThrow(fastify.swagger)
   })
 })
+
+test('transform should work with an AsyncFunction for openapi', t => {
+  t.plan(2)
+  const fastify = Fastify()
+
+  fastify.register(fastifySwagger, {
+    openapi: {},
+    transform: async schema => valid.transform(schema)
+  })
+
+  fastify.setValidatorCompiler(({ schema }) => Joi.validate(schema))
+  fastify.get('/example', opts, () => {})
+
+  fastify.ready(err => {
+    t.error(err)
+    t.resolves(fastify.swagger)
```

Comment:
You need to check if the output is valid or not.

Reply:
Done 👍